lib/widgets/drive/index.coffee

```diff
@@ -0,0 +1,119 @@
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+###
+
+_ = require('lodash')
+async = require('async')
+Promise = require('bluebird')
+drivelist = Promise.promisifyAll(require('drivelist'))
+form = require('resin-cli-form')
+InquirerList = require('inquirer/lib/prompts/list')
+UI = require('inquirer/lib/ui/baseUI')
+DriveScanner = require('./drive-scanner')
+
+_format = '<%= device %> (<%= size %>) - <%= description %>'
+
+getDrives = ->
+	drivelist.listAsync().then (drives) ->
+		Promise.fromNode (callback) ->
+			async.reject drives, drivelist.isSystem, (results) ->
+				return callback(null, results)
+
+cleanupList = (previous,currrent) ->
+	removedDrive =
+		name: _.template(_format)(currrent)
+		value: currrent.device
+
+	return _.reject previous, (drive) ->
+
+		# The reason we use _.pick and pass a custom object in _.isEqual is due to
+		# the fact the initial objects (previous, current) originate from different
+		# prototypes i.e. previous from Inquirer's Choice class and
+		# current/removedDrive from the results provided by drivelist. This makes
+		# a direct deep comparison between them impossible due the different
+		# properties that they contain.
+		return _.isEqual(_.pick(drive, 'name', 'value'), removedDrive)
+
+###*
+# @summary Prompt the user to select a drive device
+# @name visuals.drive
+# @function
+# @public
+# @memberof visuals
+#
+# @description
+# The dropdown detects and autorefreshes itself when the drive list changes.
+#
+# @param {String} [message='Select a drive'] - message
+# @returns {Promise<String>} device path
+#
+# @example
+# visuals.drive('Please select a drive').then (drive) ->
+# 	console.log(drive)
+###
+module.exports = (message = 'Select a drive') ->
+	getDrives().then (drives) ->
+		options =
+			message: message
+
+			# Even though the name property doesn't have an actual use in the code
+			# Inquirer forces us to declare it. Skipping an explicit choices
+			# declaration would cause an Error during the InquirerList instantiation.
+			name: 'drives'
+			choices: _.map drives, (drive) ->
+				return {
+					name: _.template(_format)(drive)
+					value: drive.device
+				}
+
+		ui = new UI
+			input: process.stdin
+			output: process.stdout
+		list = new InquirerList(options, ui.rl)
+		scanner = new DriveScanner getDrives,
+			interval: 1000
+			drives: drives
+
+		scanner.on 'add', (drive) ->
+
+			# New data about drives are automatically being added to both
+			# list.opt.choices.choices and list.opt.choices.realChoices through the
+			# use of push().
+			list.opt.choices.push
+				name: _.template(_format)(drive)
+				value: drive.device
+			list.render()
+
+		scanner.on 'remove', (drive) ->
+			list.opt.choices.choices = cleanupList(list.opt.choices.choices, drive)
```

Comment:
Can you add a comment above these two lines describing that we need to remove from both `choices` and `realChoices`?